multi_physics/QED/QED_table_generator/table_generator.cpp

```diff
@@ -198,71 +298,440 @@ void generate_quantum_sync_photem_table(
+    if(s_fmin != args.end())
+        params.frac_min = static_cast<RealType>(std::stod(s_fmin->second));
+
+    auto s_csize = args.find(CMD_CHI_SIZE);
+    if(s_csize != args.end())
+        params.chi_size = std::stoi(s_csize->second);
+
+    auto s_fsize = args.find(CMD_FRAC_SIZE);
+    if(s_fsize != args.end())
+        params.frac_size = std::stoi(s_fsize->second);
+
+    return params;
 }
 
+/**
+* Wraps the call to Quantum Synchrotron lookup table generators and prints some useful pieces
+* of information on screen
+*
+* @tparam RealType the floating point type to be used
+* @tparam Policy the table generation policy (it can be used to force calculations in double precision with a single precision table)
+* @param[in] params the parameters to be used to generate the tables
+* @param[in] file_name_prefix the prefix of the output file
+*/
+template<
+    typename RealType,
+    px_qs::generation_policy Policy>
+void do_quantum_sync(QuantumSyncTableParams<RealType> params, const std::string& file_name_prefix)
+{
+    std::cout << " ***** QED table generator: Quantum Synchrotron tables *****" << std::endl;
+#ifdef PXRMP_TABLE_GEN_HAS_OPENMP
+    std::cout << " Using " << omp_get_max_threads() << " OpenMP threads." << std::endl;
+#else
+    std::cout << " No OpenMP support." << std::endl;
+#endif
+
+    if (std::is_same<RealType, double>::value){
+        std::cout << " Tables will be generated in double precision." << std::endl;
+    }
+    else{
+        const auto prec =
+            (Policy == px_qs::generation_policy::force_internal_double) ?
+            "double" : "single";
+        std::cout << " Tables will be calculated in " << prec <<
+            " precision and table will be saved in single precision." << std::endl;
+    }
+
+    std::cout << " Table parameters:\n"
+        << "  - chi min   : " << params.chi_min << "\n"
+        << "  - chi_max   : " << params.chi_max << "\n"
+        << "  - frac_min  : " << params.frac_min << "\n"
+        << "  - chi_size  : " << params.chi_size << "\n"
+        << "  - frac_size : " << params.frac_size << "\n";
+
+    const auto dndt_params =
+        px_qs::dndt_lookup_table_params<RealType>{
+            params.chi_min,
+            params.chi_max,
+            params.chi_size};
+
+    const auto pair_prod_params =
+        px_qs::photon_emission_lookup_table_params<RealType>{
+            params.chi_min,
+            params.chi_max,
+            params.frac_min,
+            params.chi_size,
+            params.frac_size};
+
+    generate_quantum_sync_dndt_table<RealType, Policy>(dndt_params, file_name_prefix);
+    generate_quantum_sync_photem_table<RealType, Policy>(pair_prod_params, file_name_prefix);
+
+    std::cout << " ____________________________" << std::endl;
+}
 
-int main()
+/**
+* Prints an help message
+*/
+void print_help_message()
 {
+    std::cout << " ***** QED table generator HELP *****" << std::endl;
+
+    std::cout << std::setw(MAX_CMD_SIZE) << CMD_HELP_S <<
```

Comment:
It looks like `MAX_CMD_SIZE` is only used inside this function, so maybe it can be declared here as well?

Reply:
You are right

Reply:
It should be fixed now...